File: TransCoder_tokenized_test_set_functions/java/SMALLEST_SUBARRAY_WITH_ALL_OCCURRENCES_OF_A_MOST_FREQUENT_ELEMENT.java
static void smallestSubsegment ( int a [ ], int n ) {
  HashMap < Integer, Integer > left = new HashMap < Integer, Integer > ( );
  HashMap < Integer, Integer > count = new HashMap < Integer, Integer > ( );
  int mx = 0;
  int mn = - 1, strindex = - 1;
  for ( int i = 0;
  i < n;
  i ++ ) {
    int x = a [ i ];
    if ( count . get ( x ) == null ) {
      left . put ( x, i );
      count . put ( x, 1 );
    }
    else count . put ( x, count . get ( x ) + 1 );
    if ( count . get ( x ) > mx ) {
      mx = count . get ( x );
      mn = i - left . get ( x ) + 1;
      strindex = left . get ( x );
    }
    else if ( ( count . get ( x ) == mx ) && ( i - left . get ( x ) + 1 < mn ) ) {
      mn = i - left . get ( x ) + 1;
      strindex = left . get ( x );
    }
  }
  for ( int i = strindex;
  i < strindex + mn;
  i ++ ) System . out . print ( a [ i ] + " " );
}

File: TransCoder_tokenized_test_set_functions/cpp/SMALLEST_SUBARRAY_WITH_ALL_OCCURRENCES_OF_A_MOST_FREQUENT_ELEMENT.cpp
void smallestSubsegment ( int a [ ], int n ) {
  unordered_map < int, int > left;
  unordered_map < int, int > count;
  int mx = 0;
  int mn, strindex;
  for ( int i = 0;
  i < n;
  i ++ ) {
    int x = a [ i ];
    if ( count [ x ] == 0 ) {
      left [ x ] = i;
      count [ x ] = 1;
    }
    else count [ x ] ++;
    if ( count [ x ] > mx ) {
      mx = count [ x ];
      mn = i - left [ x ] + 1;
      strindex = left [ x ];
    }
    else if ( count [ x ] == mx && i - left [ x ] + 1 < mn ) {
      mn = i - left [ x ] + 1;
      strindex = left [ x ];
    }
  }
  for ( int i = strindex;
  i < strindex + mn;
  i ++ ) cout << a [ i ] << " ";
}